dna/dna.go

```diff
@@ -1,38 +1,184 @@
+// Package dna provides functionality to create and manipulate DNA sequences.
 package dna
 
-import ()
+import "strings"
 
+// Base stores a single nucleotide as a byte.
 type Base byte
 
-// TODO: change these names so that all variables can be seen
-// from outside.  Maybe BaseA, Basea, etc
 const (
-	A   Base = 0
-	C   Base = 1
-	G   Base = 2
-	T   Base = 3
-	N   Base = 4
-	a   Base = 5
-	c   Base = 6
-	g   Base = 7
-	t   Base = 8
-	n   Base = 9
-	Gap Base = 10
-	Dot Base = 11
+	A      Base = 'A'
```

Comment:
What do we gain by using ascii values instead of values counting up from zero?  I think this going to have some ripple effects where conversion to using less bits are not as straight forward and that bases can not be used as indices into slices (like for a scoring matrix).  Not against this change if there are some strong benefits, but I think it will have some minor drawbacks.